region/brinstar/pink/Pink Brinstar Hopper Room.json

```diff
@@ -812,6 +854,71 @@
       "reusableRoomwideNotable": "Pink Hoppers Shinespark into the Wall to Ride the Elevator",
       "note": "Shinesparking horizontally into the wall will trigger the elevator as Samus's echos hit it."
     },
+    {
+      "link": [2, 3],
+      "name": "Pink Hoppers X-Ray Climb Spring Ball Jump",
+      "entranceCondition": {
+        "comeInWithDoorStuckSetup": {}
+      },
+      "requires": [
+        "canXRayClimb",
+        "canTrickyJump",
+        "canLateralMidAirMorph",
+        "canTrickySpringBallJump",
+        "canSpringFling"
+      ],
+      "note": [
+        "X-Ray climb until Samus is a little over halfway off screen; the position is not precise.",
+        "Face right, then do a turnaround buffer spin jump into a Spring Ball jump to reach the ledge.",
+        "The pause input can be pressed at the same time as the turnaround.",
+        "It helps to do an airball to gain horizontal speed.",
+        "The jump input after pausing should not be buffered, in order to allow a little time for the spring fling horizontal speed to take effect."
+      ]
+    },
+    {
+      "link": [2, 3],
+      "name": "Pink Hoppers X-Ray Climb HiJump Insane Jump",
+      "entranceCondition": {
+        "comeInWithDoorStuckSetup": {}
+      },
+      "requires": [
+        "HiJump",
+        "canXRayClimb",
+        "canInsaneJump"
+      ],
+      "note": [
+        "Get stuck 3 pixels inside the door, and X-ray climb to a precise height:",
```

Comment:
> 3 pixels inside the door

Is it not a normal doorstuck?

Reply:
It's a "normal" doorstuck setup as far as I know, but that setup can end up getting you stuck either 1, 2, or 3 pixels in the door, and this strat needs 3. I assume there must be some subtlety to the exact movement to get 3 pixels vs. other values but I don't understand it.

Reply:
Different run speeds and transition heights can work... so I'd say it's a combination of the two, as well as fall speed, to land on the correct pixel position.  So RNG to me.

Slightly easier than a `canRightSideDoorStuckFromWater`.  The jump from the wall is a textbook `canInsaneJump`.  Running for 2 pixels to get the run speed takes a bit of practice to get the feel for.

Reply:
There is some concern over blind xray standups since it is often hard to tell if you missed one.  Sometimes you can tell if the crouch feels wrong, but we do want a level of comfort with Xray Climbing when blind counting.

Reply:
Do you think we need a new tech for the specific doorstuck setup? The "canInsaneJump" already gates these strats to a high difficulty level. I also just added a "canOffScreenMovement" requirement for the parts of counting your climbs (and knowing if you missing any)? The longer blind climb at least has some lenience, where you could shoot for the high number of climbs (22), and if you miss a couple (e.g. only get 20 climbs) it will still work.

Reply:
I think there does need to be a tech for describing getting stuck with a particular pixel value.  `doorStuckFromWater` I'd like to leave as it's own thing.  GMode doorstuck is part of GMode, but knowledge of which pixels are good for climbing could be under the new tech.

I'd leave this as it is for this case, but I'd guess you can run from a wall in more rooms like East Cac Alley.  GT's room would be 🤡.  Maybe not too many places.  I'm guessing you need `Normal` physics and right side door stuck for the run to work.  

Turnaround spinjumps like the springball strat would use basic door stuck tech.